Build wheels with rustls

ekroon opened the pull request:
The upgrade that came with the SIMD build in the Dockerfile #81, broke the `columnq` and `roapi-http` builds. I tried fixing this in multiple ways but in the end upgrading the container to a more recent version and using rustls instead of native-tls (openssl) seems to work. I am not sure if my solution is acceptable, but I am not sure how to fix it otherwise.

I tried combining this with #31 but I only could get the x86_64 build to work.

houqp commented on .github/workflows/columnq_cli_release.yml:
```diff
@@ -140,7 +142,7 @@ jobs:
       - name: Build Wheels
         run: |
-          pip3 install 'maturin<0.12'
+          sudo python3 -m pip install maturin
```
could you still pin maturin to <0.12?

houqp commented on .github/workflows/columnq_cli_release.yml:
```diff
@@ -3,10 +3,11 @@ name: columnq-cli release
       - 'columnq-cli/**'
       - 'Cargo.lock'
-    branches: [ main ]
+    branches: [ main, rustls ]
```
i believe rustls was added for testing purpose?

ekroon replied:
definitely :)

houqp commented on .github/workflows/roapi_http_release.yml:
```diff
@@ -3,10 +3,11 @@ name: roapi-http release
       - 'roapi-http/**'
       - 'Cargo.lock'
-    branches: [ main ]
+    branches: [ main, rustls ]
```
same here, i think rustls can be removed here.

houqp reviewed (commented):
Thanks @ekroon taking care of this! I think it's fine to switch to rustls as the default option, but we should still try to support native-tls if possible. Have you tried reverting the nightly toolchain version back to `2021-07-17`?

ekroon commented:
> Thanks @ekroon taking care of this! I think it's fine to switch to rustls as the default option, but we should still try to support native-tls if possible. Have you tried reverting the nightly toolchain version back to `2021-07-17`?

What I tried and didn't get to work:

- Revert nightly toolchain back to '2021-07017'. The SIMD build fails, so we would not have a SIMD build anymore
- Build without rustls on the previous (benfred/rust-musl-cross) container, (this is the first situation  that fails)
- Build with rustls on x86_64 gnu (glibc) cross container through (https://github.com/messense/maturin-action), which only worked for the x86_64 build, the rest failed with problems linking `renameat2` (which is someting in the `deltalake` crate I think)
- Build without rustls on the new container in the current PR, but this container doesn't have OpenSSL installed, so no bueno of course. (The repository mentions: https://github.com/rust-embedded/cross/issues/229, to depend on openssl-src, which would be a way to support musl + openssl I think)
- Build with rustls on the new container in the current PR, and that is what you are seeing here.

I have to say, that there might be something to circle back to here, but I don't think that reverting back to an older nightly is the solution. 

Adding native-tls back behind as a feature sound definitely as a good idea. Do you want it included in this PR?

houqp commented:
yes, it would be great if we can add native-tls as a feature, some users might prefer being able to patch ssl bugs in production without having to rebuild the binaries.

houqp commented:
I have to say it's quite strange that SIMD build fails with `2021-07017` nightly though because it's been working outside of docker. But I agree with you that it would be better if we can fix it forward.

ekroon commented:
> yes, it would be great if we can add native-tls as a feature, some users might prefer being able to patch ssl bugs in production without having to rebuild the binaries.

Yes definitely agree. 

> I have to say it's quite strange that SIMD build fails with `2021-07017` nightly though because it's been working outside of docker. But I agree with you that it would be better if we can fix it forward.

This one would be easy to double-check by lowering this dependency again and check the build, I created #84

ekroon commented:
Results for the builds are here: https://github.com/ekroon/roapi/actions?query=branch%3Arustls

ekroon commented:
@houqp I added the feature for `native-tls-vendored` which uses vendored openssl on Linux. If someone wants to use the system provided openssl they can use the `OPENSSL_NO_VENDOR` environment variabel as described [here](https://docs.rs/openssl/0.10.36/openssl/#manual).

The `yup-oauth2` dependency defaulted to rustls for hyper, so the previous builds depended on both, I fixed that too.

I added the new features to the linux build-matrix and the windows and macos build only build the default `rustls` feature. Only the `rustls` matrix-items will have their wheels and artifacts publiced.

houqp commented:
Thank you @ekroon for taking care of all the build issues :D
